crypto/objects/o_names.c

```diff
@@ -157,11 +174,13 @@ const char *OBJ_NAME_get(const char *name, int type)
 {
     OBJ_NAME on, *ret;
     int num = 0, alias;
+    const char *value = NULL;
 
     if (name == NULL)
         return (NULL);
-    if ((names_lh == NULL) && !OBJ_NAME_init())
+    if (!OBJ_NAME_init())
         return (NULL);
+    CRYPTO_THREAD_read_lock(lock);
```

Comment:
It looks like the tracked statistics are just a couple of counters, so we could swap over to atomic operations and avoid write locking.